src/main/java/org/folio/rest/impl/OrdersImpl.java

```diff
@@ -386,4 +389,85 @@ public void deleteOrdersPiecesById(String pieceId, String lang, Map<String, Stri
       .thenAccept(ok -> asyncResultHandler.handle(succeededFuture(deletePieceHelper.buildNoContentResponse())))
       .exceptionally(fail -> handleErrorResponse(asyncResultHandler, deletePieceHelper, fail));
   }
+
+  @Override
+  @Validate
+  public void postOrdersAcquisitionsUnitAssignments(String lang, AcquisitionsUnitAssignment entity, Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    AcquisitionsUnitAssignmentsHelper helper = new AcquisitionsUnitAssignmentsHelper(okapiHeaders, vertxContext, lang);
+
+    helper.createAcquisitionsUnitAssignment(entity)
+      .thenAccept(unit -> {
+        if (logger.isInfoEnabled()) {
+          logger.info("Successfully created new acquisitions unit: " + JsonObject.mapFrom(unit).encodePrettily());
+        }
+
+        asyncResultHandler.handle(succeededFuture(helper.buildResponseWithLocation(String.format(ACQUISITIONS_UNIT_ASSIGNMENTS_LOCATION_PREFIX, unit.getId()), unit)));
+        helper.closeHttpClient();
```

Comment:
public Response buildResponseWithLocation(String endpoint, Object body) already execute http-client closing. No need to explicitly execute closeHttpClient().

Reply:
Good point. Thank you. I already fixed this.